wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,418 @@
+from discord.ext import commands
+import discord
+from resources.utilities.embed import embed as em
+from WalleModels.models import BanRecords
+import datetime
+import pytz
+from typing import Union
+import logging
+logger = logging.getLogger('wall_e')
+
+
+class Ban(commands.Cog):
+
+    def __init__(self, bot, config):
+        self.bot = bot
+        self.config = config
+        self.ban_list = []
+        self.error_colour = 0xA6192E
+        self.mod_channel = None
+
+    @commands.Cog.listener(name='on_ready')
+    async def load(self):
+        """Grabs channel to send mod reports to and reads in the ban_list from db"""
+
+        mod_channel_name = self.config.get_config_value('basic_config', 'MOD_CHANNEL')
+        if self.config.get_config_value('basic_config', 'ENVIRONMENT') != 'PRODUCTION':
+            await self.make_mod_channel()
+        else:
+            logger.info(f'[Ban load()] Attempting to get the report channel: {mod_channel_name}')
+
+            self.mod_channel = discord.utils.get(self.bot.guilds[0].channels, name=mod_channel_name)
+            if self.mod_channel:
+                logger.info(f"[Ban load()] #{mod_channel_name} channel successfully found: {self.mod_channel}")
+            else:
+                logger.info(f"[Ban load()] Couldn't get {mod_channel_name} from guild."
+                            " Channel doesn't exist. Exiting.")
+                exit('No mod channel')
```

Comment:
you will want to sleep for like 20 seconds before printing this out. otherwise, the log above it doesn't get a chance to get printed to discord